### PR TITLE
Integrate pub schemas

### PR DESCRIPTION
**Pull request by CMClay**
Public data info needs to be represented in the schemas for our API explorer and docs.

##### Resources:
- assessments
- parcels
- transaction

### CODE REVIEW

**Comment by CMClay** [comment]
@wired8 @lvivier @MadManMathew  Can one of you pub dudes provide me with a short description of the non-obvious fields in the pub schemas ie. lotTopography, lotSizeIrregular...

File: lib/assessment-desc.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+module.exports = {
+  id: 'Unique ID to identify the assessment.',
+  apn: '', //TODO
```
Comment (CMClay) [inline lib/assessment-desc.js]: @wired8 Can you please help me out and give a small description of some these fields from pub. You can just add line comments here. Thanks!

Comment (wired8) [inline lib/assessment-desc.js]: @lvivier is there a data dictionary for these, or anything in MSSQL that perhaps describes what these are?

Comment (lvivier) [inline lib/assessment-desc.js]: `apn` is Assessor Parcel Number, the unique identifier assigned by the tax authority

File: lib/assessment-desc.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  fips: 'Five digit county code ID.',
+  county: 'County name.',
+  state: 'Two character state abbreviation.',
+  censusTract: '', //TODO
+
+  taxID: '', //TODO
+  taxAmount: '', //TODO
+  taxYear: '', //TODO
+
+  taxExemption: '', //TODO
```
Comment (lvivier) [inline lib/assessment-desc.js]: Type of tax exemption, if any

File: lib/assessment-desc.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+module.exports = {
+  id: 'Unique ID to identify the assessment.',
+  apn: '', //TODO
+  fips: 'Five digit county code ID.',
+  county: 'County name.',
+  state: 'Two character state abbreviation.',
+  censusTract: '', //TODO
+
+  taxID: '', //TODO
+  taxAmount: '', //TODO
```
Comment (lvivier) [inline lib/assessment-desc.js]: Total dollar amount of property tax

File: lib/assessment-desc.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+module.exports = {
+  id: 'Unique ID to identify the assessment.',
+  apn: '', //TODO
+  fips: 'Five digit county code ID.',
+  county: 'County name.',
+  state: 'Two character state abbreviation.',
+  censusTract: '', //TODO
```
Comment (lvivier) [inline lib/assessment-desc.js]: Census tract is a geographic region used by the US Census Bureau

File: lib/assessment-desc.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+  taxID: '', //TODO
```
Comment (lvivier) [inline lib/assessment-desc.js]: Unique identifier assigned for taxation purposes (different from APN)

File: lib/assessment-desc.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  taxAmount: '', //TODO
+  taxYear: '', //TODO
```
Comment (lvivier) [inline lib/assessment-desc.js]: Calendar year corresponding to the tax amount. May be different from `year` b/c some counties do not reassess every year

File: lib/assessment-desc.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+  taxExemption: '', //TODO
+  year: 'Assessment year.',
+  landValue: 'Assessed value of land.',
+  improvementValue: 'Assessed value of improvements', //TODO,
+  totalValue: 'Total assessed value of land and improvements.',
+  landUseGeneral: '', //TODO
```
Comment (lvivier) [inline lib/assessment-desc.js]: Two character land use code (example: `"RR"` for Residential, `"AG"` for Agricultural)

File: lib/assessment-desc.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  landUseCode: '', //TODO
```
Comment (lvivier) [inline lib/assessment-desc.js]: Five character code describing this parcel's land use type

File: lib/assessment-desc.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+  landUseDescription: '', //TODO
```
Comment (lvivier) [inline lib/assessment-desc.js]: Description that matches the land use code

File: lib/assessment-desc.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  zoningCode: '', //TODO
```
Comment (lvivier) [inline lib/assessment-desc.js]: Municipality-specific zoning code describes what part of the zoning or land use bylaw applies to this parcel

File: lib/assessment-desc.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+  zoningDescription: '', //TODO
+  numberOfBuildings: 'Number of buildings.',
+  lotSizeAcres: 'Size of the lot in acres.',
+  lotSizeSquareFeet: 'Size of the lot in square feet.',
+  lotSizeFrontage: 'Distance in feet between the front of the building and the property boundary',
```
Comment (lvivier) [inline lib/assessment-desc.js]: Frontage is actually the width of the lot at the street (or shore for water access pptys)

File: lib/assessment-desc.js [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+  lotSizeDepth: '', //TODO
```
Comment (lvivier) [inline lib/assessment-desc.js]: Average depth of lot perpendicular to frontage

File: lib/assessment-desc.js [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+  lotSizeIrregular: '', //TODO
+  lotTopography: '', //TODO
```
Comment (lvivier) [inline lib/assessment-desc.js]: Short description of lot topography

File: lib/assessment-desc.js [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+  landUseCode: '', //TODO
+  landUseDescription: '', //TODO
+  zoningCode: '', //TODO
+  zoningDescription: '', //TODO
+  numberOfBuildings: 'Number of buildings.',
+  lotSizeAcres: 'Size of the lot in acres.',
+  lotSizeSquareFeet: 'Size of the lot in square feet.',
+  lotSizeFrontage: 'Distance in feet between the front of the building and the property boundary',
+  lotSizeDepth: '', //TODO
+  lotSizeIrregular: '', //TODO
```
Comment (lvivier) [inline lib/assessment-desc.js]: Perimeter measurements for non-rectangular shaped lots

File: lib/transaction-desc.js [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+  countyTransferTax: 'County transfer tax.',
+  stateTransferTax: 'County transfer tax.',
+  totalTransferTax: 'Total transfer tax.',
+  intraFamilyTransfer: 'Flag for intra family transfers.', //TODO
+  transferTaxExempt: 'Flag for transfer tax exemption.', //TODO
+  buyerName: 'Name of buyer.',
+  sellerName: 'Name of seller.',
+  borrowerVesting: '', //TODO
+  lenderName: 'Name of lender.',
+  lenderType: '', //TODO
```
Comment (lvivier) [inline lib/transaction-desc.js]: The type of institution (bank, private lender, etc)

File: lib/assessment-desc.js (hunk 13 above)
Comment (CMClay) [inline lib/assessment-desc.js]: Thanks @lvivier!

File: lib/transaction-desc.js [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+  recordingPageNumber: '', //TODO
+  documentTypeCode: '', //TODO
+  documentType: 'Type of document', //TODO
+  documentDate: '', //TODO
+  signatureDate: 'Date transaction was signed.',
+  effectiveDate: 'Effective date transaction takes effect.',
+  buyerVesting: 'Vesting process for buyer.',
+  buyerMultiVesting: 'Flag for multi vesting.',
+  partialInterestTransfer: '', //TODO
+  partialInterestTransferPercent: '', //TODO
```
Comment (CMClay) [inline lib/transaction-desc.js]: @lvivier Just a few fields left in the transaction. Thanks for your help.

**Comment by CMClay** [comment]
@lvivier @wired8 Review please.

**Comment by wired8** [comment]
:+1:

**Comment by CMClay** [comment]
@CalvinScott Can you review this for me please.

File: lib/assessment-desc.js [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+  totalValue: 'Total assessed value of land and improvements.',
+  landUseGeneral: 'Two character land use code (example: "RR" for Residential, "AG" for Agricultural).',
+  landUseCode: 'Five character code describing this parcel\'s land use type.',
+  landUseDescription: 'Description that matches the land use code.',
+  zoningCode: 'Municipality-specific zoning code indicates what part of the zoning or land use bylaw applies to this parcel.',
+  zoningDescription: 'Short description of zoning code.',
+  numberOfBuildings: 'Number of buildings.',
+  lotSizeAcres: 'Size of the lot in acres.',
+  lotSizeSquareFeet: 'Size of the lot in square feet.',
+  lotSizeFrontage: 'Eidth of the lot at the street (or shore for properties with water access).',
```
Comment (CalvinScott) [inline lib/assessment-desc.js]: Eidth = width?

Comment (CMClay) [inline lib/assessment-desc.js]: Thanks.

File: lib/assessment-desc.js [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+  lotSizeDepth: 'Average depth of lot perpendicular to frontage.',
+  lotSizeIrregular: 'Perimeter measurements for non-rectangular shaped lots.',
+  lotTopography: 'Short description of lot topography.',
+  building: 'Building details.',
+  ownerName: 'Name of owner.',
+  address: 'Object conatining address attributes.',
+  legal: 'Legal details of the assessment.',
+  coordinates: 'Latitude and longitude.'
+}
```
Comment (CalvinScott) [inline lib/assessment-desc.js]: Semicolon?

**Comment by CalvinScott** [comment]
Looks good :+1:

**Comment by CMClay** [comment]
Thanks @CalvinScott !

**Pull request merged**